File: docs/07-archive/legacy-repos/ebs_bo/src/bo/schemas/competition.py
from pydantic import BaseModel


class CompetitionCreate(BaseModel):
    name: str
    competition_type: int = 0
    competition_tag: int = 0


class CompetitionUpdate(BaseModel):
    name: str | None = None
    competition_type: int | None = None
    competition_tag: int | None = None


class CompetitionRead(BaseModel):
    competition_id: int
    name: str
    competition_type: int
    competition_tag: int
    created_at: str
    updated_at: str

File: team1-frontend/lib/models/entities/competition.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'competition.freezed.dart';
part 'competition.g.dart';

@freezed
class Competition with _$Competition {
  const factory Competition({
    @JsonKey(name: 'competitionId') required int competitionId,
    required String name,
    @JsonKey(name: 'competitionType') required int competitionType,
    @JsonKey(name: 'competitionTag') required int competitionTag,
    @JsonKey(name: 'createdAt') required String createdAt,
    @JsonKey(name: 'updatedAt') required String updatedAt,
  }) = _Competition;

  factory Competition.fromJson(Map<String, dynamic> json) =>
      _$CompetitionFromJson(json);
}
